clients/ui/bff/internal/api/middleware.go

```diff
@@ -83,11 +84,17 @@ func resolveBearerToken(k8s integrations.KubernetesClientInterface, header http.
 	return bearerToken, nil
 }
 
-func resolveModelRegistryURL(id string, client integrations.KubernetesClientInterface) (string, error) {
-	serviceDetails, err := client.GetServiceDetailsByName(id)
+func resolveModelRegistryURL(id string, app *App) (string, error) {
```

Comment:
I'm not a fan of passing a pointer to the app struct into this function. It's meant as a utility function so I would stick to value semantics and add params for each value it requires.

An alternative would be to convert this function into a method of the app struct and use a receiver param to access the methods.

Reply:
Agree with Alex here.